src/main/java/com/google/sps/data/User.java

```diff
@@ -100,106 +97,92 @@ public String getId() {
-
-  public ArrayList<Long> getCookbook() {
-    return getPropertyArrayList("cookbook");
+  public List<Key> getCookbookList() {
+    return getRecipeList(COOKBOOK);
   }
 
-  public ArrayList<Long> getUserRecipes() {
-    return getPropertyArrayList("userRecipes");
+  public List<Key> getPlannerList() {
+    return getRecipeList(PLANNER);
   }
 
-  public ArrayList<Long> getPlanner() {
-    return getPropertyArrayList("planner");
+  public List<Key> getDraftList() {
+    return getRecipeList(DRAFTS);
   }
 
-  public void addRecipeToPlanner(long id) {
-    addIdToRecipeList(id, "planner");
+  public void addCookbookKey(Key key) {
+    addKey(key, COOKBOOK);
   }
 
-  public void addRecipeToCookbook(long id) {
-    addIdToRecipeList(id, "cookbook");
+  public void addPlannerKey(Key key) {
+    addKey(key, PLANNER);
   }
 
-  public void addRecipeToUserRecipes(long id) {
-    addIdToRecipeList(id, "userRecipes");
+  public void addDraftKey(Key key) {
+    addKey(key, DRAFTS);
   }
 
-  public void removeRecipeFromPlanner(long id) {
-    removeIdFromRecipeList(id, "planner");
+  public void removeCookbookKey(Key key) {
+    removeKey(key, COOKBOOK);
   }
 
-  public void removeRecipeFromCookbook(long id) {
-    removeIdFromRecipeList(id, "cookbook");
+  public void removePlannerKey(Key key) {
+    removeKey(key, PLANNER);
   }
 
-  public void removeRecipeFromUserRecipes(long id) {
-    removeIdFromRecipeList(id, "userRecipes");
+  public void removeDraftKey(Key key) {
+    removeKey(key, DRAFTS);
   }
 
+  // PUBLIC METHODS END HERE
+
   /**
-   * Helper method for removing recipe IDs form specific user recipe list.
-   * @Param id recipe ID to be removed from current User entity.
-   * @Param recipeListType Property name of the list to removed from.
+   * Adds the current version of User entity to datastore.
    */
-  private void removeIdFromRecipeList(long id, String recipeListType) {
-    Gson gson = new Gson();
-    ArrayList<Long> list = getPropertyArrayList(recipeListType);
-    list.remove(new Long(id));
-    entity.setProperty(recipeListType, gson.toJson(list));
-    putEntity();
+  private void uploadEntity() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(entity);
   }
 
   /**
-   * Helper method for adding recipe IDs to specific user recipe list.
-   * @Param id recipe ID to be added to current User entity.
-   * @Param recipeListType Property name of the list to add to.
+   * Helper method for retrieving ArrayList<Key> of given recipe list.
    */
-  private void addIdToRecipeList(long id, String recipeListType) {
-    Gson gson = new Gson();
-    ArrayList<Long> list = getPropertyArrayList(recipeListType);
-    if (!list.contains(id)) {
-      list.add(id);
+  @SuppressWarnings("unchecked") // Compiler cannot verify generic property type.
+  private ArrayList<Key> getRecipeList(String listName) {
+    ArrayList<Key> arrayListValue = (ArrayList<Key>) entity.getProperty(listName);
```

Comment:
Is there a better name for what this is? Maybe recipeListValue or something that indicates it's a list of recipe keys?